torch/jit/__init__.py

```diff
@@ -1573,115 +1572,94 @@ def __getstate__(self):
+    an nn.Module into a fresh ScriptModule
+    """
+    script_module = ScriptModule()
+
+    # A quick check that the module's super() is called so everything is set up
+    # for this method to do the copying it needs to
+    if not hasattr(module, '_parameters'):
+        raise RuntimeError("'{}' has not been initialized, did you forget to call 'super()'?"
+                           .format(type(module).__name__))
+
+
+    # HACK: To support None nn.Parameters and buffers, don't copying them if they
+    # are both in __constants__ and are of a None value. If they are not none,
+    # copy them as usual. For this reason we need to know the constants here
+    # but not actually set it on the script_module until after the Parameters
+    # and buffers are copied.
+    constants_set = set(getattr(module, "__constants__", []))
+
+    # Copy attributes
+    for name in dir(module):
+        value = getattr(module, name)
+        if isinstance(value, Attribute):
+            setattr(script_module, name, item)
+
+    def copy_tensors(the_list, setter):
+        for name in the_list:
+            value = getattr(module, name)
+            if value is None:
+                # For None buffers/parameters, don't register it (this path treats it as a
+                # Optional[Tensor] constant where the value is None)
+                object.__setattr__(module, name, None)
+            else:
+                setter(name, value)
+
+    # Copy buffers
+    copy_tensors(
+        module._buffers,
+        lambda name, value: script_module.register_buffer(name, module._buffers[name])
+    )
+
+    # Copy parameters
+    copy_tensors(
+        module._parameters,
+        lambda name, value: setattr(script_module, name, value)
+    )
+
+    # Copy submodules
+    for name in module._modules:
+        value = getattr(module, name)
+        setattr(script_module, name, value)
+
+    # Constants annotated via `Final[T]` rather than being added to `__constants__`
+    for name, ann in getattr(module, '__annotations__', {}).items():
+        if torch._jit_internal.is_final(ann):
+            constants_set.add(name)
+
+    # Copy constants
+    script_module.__dict__["_constants_set"] = constants_set
+    for name in constants_set:
+        if hasattr(module, name):
+            value = getattr(module, name)
+            if (name in module._parameters or name in module._buffers) and value is not None:
+                # for 'None' parameters/buffers, don't actually add their values if it exists
+                continue
+            ScriptModule.__setattr__(script_module, name, getattr(module, name))
+
+    # Copy overloads
+    script_module.__dict__["_overloads"] = dict(getattr(module, "__overloads__", {}))
+
+    # Copy python ops
+    for name in dir(module):
+        if hasattr(script_module, name):
+            # Skip Python class stuff and don't re-assign anything
+            continue
+        value = getattr(module, name)
```

Comment:
Is this for `@ignore` functions? We should assert that if it's the case

Reply:
We want to take all functions / module attributes here so that we can lazily compile them without needing to keep the original class around